sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -846,6 +857,15 @@ abstract class ToTimestamp
               case NonFatal(_) => null
             }
           }
+        case LongType =>
+          val input = t.asInstanceOf[Long]
+          if ( minRange < input && input < maxRange ) {
+            Math.multiplyExact(input, scaleFactor.asInstanceOf[Long])
```

Comment:
`Math.multiplyExact` throws `ArithmeticException`. You don't need to explicitly check the ranges.

Reply:
My earlier concern was about whether we support the full range of Long as valid timestamps, or if the actual supported range is smaller. I googled a bit, and it seems that the full range is supported. So that's fine then without checks.